.travis.yml

```diff
@@ -57,28 +59,55 @@ matrix:
             rm -rf "$TAPS/caskroom/homebrew-cask";
           fi;
           if [ ! -f $HOME/.osx_cache/cached ]; then
-            brew tap homebrew/homebrew-cask;
+            echo "==> [Not found cache] brew tap homebrew/homebrew-cask";
+            echo "[NOTE]";
+            echo "If brew is executed without HOMEBREW_NO_AUTO_UPDATE=1,";
+            echo "python3 cannot be installed, so this is added as a temporary workaround.";
+            echo "If it is xcode 9.4 or higher, clear this patch.";
+            HOMEBREW_NO_AUTO_UPDATE=1 brew tap homebrew/homebrew-cask;
           else
+            echo "==> [Found cache] HOMEBREW_NO_AUTO_UPDATE=1 brew tap homebrew/homebrew-cask";
             HOMEBREW_NO_AUTO_UPDATE=1 brew tap homebrew/homebrew-cask;
           fi
         - HOMEBREW_NO_AUTO_UPDATE=1 brew cask install osxfuse
-        - S3FS_BREW_PACKAGES='awscli cppcheck truncate';
+        - S3FS_BREW_PACKAGES='cppcheck python3';
           for s3fs_brew_pkg in ${S3FS_BREW_PACKAGES}; do
             brew list | grep -q ${s3fs_brew_pkg};
             if [ $? -eq 0 ]; then
-              brew outdated | grep -q ${s3fs_brew_pkg} && HOMEBREW_NO_AUTO_UPDATE=1 brew upgrade ${s3fs_brew_pkg};
+              brew outdated | grep -q ${s3fs_brew_pkg};
+              if [ $? -eq 0 ]; then
+                echo "==> Try to upgrade ${s3fs_brew_pkg}";
+                HOMEBREW_NO_AUTO_UPDATE=1 brew upgrade ${s3fs_brew_pkg};
+              fi
             else
+              echo "==> Try to install ${s3fs_brew_pkg}";
               HOMEBREW_NO_AUTO_UPDATE=1 brew install ${s3fs_brew_pkg};
             fi;
           done
+        - pip3 --version;
```

Comment:
Not needed with python2?

Reply:
Since 10.12 has python2 by default, I think it's fine to check only pip3.